File: website/src/main/java/app/api/AccountAJAXWebService.java
package app.api;

import app.api.user.CurrentUserAJAXResponse;
import app.api.user.LoginAJAXRequest;
import app.api.user.LoginAJAXResponse;
import core.framework.api.web.service.GET;
import core.framework.api.web.service.PUT;
import core.framework.api.web.service.Path;

public interface AccountAJAXWebService {
    @PUT
    @Path("/ajax/login")
    LoginAJAXResponse login(LoginAJAXRequest request);

    @PUT
    @Path("/ajax/logout")
    void logout();

    @GET
    @Path("/ajax/currentUser")
    CurrentUserAJAXResponse currentUser();
}

File: website-frontend/src/service/AccountAJAXWebService.ts
import {CurrentUserAJAXResponse, LoginAJAXRequest, LoginAJAXResponse} from "type/api";
import {ajax} from "core-fe";

export class AccountAJAXWebService {
    static currentUser(): Promise<CurrentUserAJAXResponse> {
        return ajax("GET", "/ajax/currentUser", {}, null);
    }
    static login(request: LoginAJAXRequest): Promise<LoginAJAXResponse> {
        return ajax("PUT", "/ajax/login", {}, request);
    }
    static logout(): Promise<void> {
        return ajax("PUT", "/ajax/logout", {}, null);
    }
}
